### PR TITLE
xt_quirk: link devel sources instead copy

### PR DESCRIPTION
**Pull request by varder**
Use symbol link for development sources instead of copy

PR should be merged with https://github.com/xen-troops/meta-xt-prod-devel/pull/327

@andr2000
@lorc

Signed-off-by: Valerii Chubar <valerii_chubar@epam.com>

### CODE REVIEW

File: classes/xt_quirks.bbclass [hunk 1]
```diff
@@ -44,18 +41,14 @@ python do_unpack_xt_extras() {
         urls.append("file://" + "build-versions.inc" + ";subdir=repo/build/conf")
 
     for url in urls:
-        type, _, location, _, _, _ = bb.fetch.decodeurl(url)
-        item = check_url_or_pack(url, type, location, d) + " "
-        d.appendVar("SRC_URI", item or "")
+        try:
+            type, _, location, _, _, _ = bb.fetch.decodeurl(url)
+            path = bb.fetch2.localpath(url, d)
+            bb.debug(1, "Make symbol link for path:" + path)
```
Comment (andr2000) [inline classes/xt_quirks.bbclass]: s/symbol/symbolic

**Review by andr2000 — APPROVED** [review]
After proper testing and nit fixed:
`Reviewed-by: Oleksandr Andrushchenko <oleksandr_andrushchenko@epam.com>`

File: classes/xt_quirks.bbclass [hunk 2]
```diff
@@ -100,3 +100,6 @@ python do_xt_config_reconstruct() {
     with open(conf_file, "a") as f:
         f.write("require" + " " + "build-versions.inc" + "\n")
 }
+
+do_configure[nostamp] = "1"
```
Comment (andr2000) [inline classes/xt_quirks.bbclass:87]: Will do_configure remove the lower Yocto and start the build from the start (with state caches used?)

File: classes/xt_quirks.bbclass [hunk 3]
```diff
@@ -44,18 +41,24 @@ python do_unpack_xt_extras() {
         urls.append("file://" + "build-versions.inc" + ";subdir=repo/build/conf")
 
     for url in urls:
-        type, _, location, _, _, _ = bb.fetch.decodeurl(url)
-        item = check_url_or_pack(url, type, location, d) + " "
```
Comment (andr2000) [inline classes/xt_quirks.bbclass:48]: check_url_or_pack is no longer needed

File: classes/xt_quirks.bbclass [hunk 4]
```diff
@@ -44,18 +41,24 @@ python do_unpack_xt_extras() {
...
-        d.appendVar("SRC_URI", item or "")
-
-    # now call real unpacker to do the rest
-    bb.build.exec_func('base_do_unpack', d)
-
-    # remove the archives we created
-    urls = (d.getVar("XT_QUIRK_UNPACK_SRC_URI") or "").split()
-    for url in urls:
-        type, _, location, _, _, _ = bb.fetch.decodeurl(url)
-        remove_tarfile(url, type, location, d)
```
Comment (andr2000) [inline classes/xt_quirks.bbclass:58]: remove_tarfile is no longer needed

File: classes/xt_quirks.bbclass (hunk 2 above)
Comment (varder) [reply classes/xt_quirks.bbclass:87]: I have tested and in this case do_configure does not cause lower yocto removal.

Comment (andr2000) [reply classes/xt_quirks.bbclass:87]: looks good then

File: classes/xt_quirks.bbclass (hunk 3 above)
Comment (varder) [reply classes/xt_quirks.bbclass:48]: done

File: classes/xt_quirks.bbclass (hunk 4 above)
Comment (varder) [reply classes/xt_quirks.bbclass:58]: done

**Comment by rshym** [comment]
Tested by build of prod-devel (with https://github.com/xen-troops/meta-xt-prod-devel/pull/327) and prod-aos (with https://github.com/xen-troops/meta-xt-prod-aos/pull/111).
Products built successfully.

**Pull request merged**